FluentTc.Tests/Locators/GitVCSRootBuilderTests.cs

```diff
@@ -38,41 +38,109 @@ public void SimpleCreation()
+
+            vcsRootBuilder
+                .AgentCleanFilePolicy(AgentCleanFilePolicy.AllIgnoredUntrackedFiles)
+                .AgentCleanPolicy(AgentCleanPolicy.Always)
+                .AuthMethod(AuthMethod.TeamcitySshKey)
+                .Branch("branch")
+                .BranchSpec("branchSpec")
+                .CheckoutSubModule()
+                .Id("id")
+                .IgnoreKnownHosts()
+                .Name("name")
+                .ProjectId("projectId")
+                .UploadedKey("keyName")
+                .Url(url)
+                .UseAlternates()
+                .UserNameStyle(UserNameStyle.AuthorName);
+
+            var vcsRoot = vcsRootBuilder.GetVCSRoot();
+            vcsRoot.Id.Should().Be("id");
+            vcsRoot.Name.Should().Be("name");
+            vcsRoot.Project.Id.Should().Be("projectId");
+            var properties = vcsRoot.Properties.Property;
+            properties.FirstOrDefault(
+                p => p.Name == "agentCleanFilePolicy" &&
+                     p.Value == "ALL_IGNORED_UNTRACKED_FILES").Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "agentCleanPolicy" &&
+                     p.Value == "ALWAYS").Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "authMethod" &&
+                     p.Value == "ANONYMOUS").Should().BeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "authMethod" &&
+                     p.Value == "TEAMCITY_SSH_KEY").Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "branch" &&
+                     p.Value == "branch").Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "teamcity:branchSpec" &&
+                     p.Value == "branchSpec").Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "submoduleCheckout" &&
+                     p.Value == "CHECKOUT").Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "ignoreKnownHosts" &&
+                     p.Value == "true").Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "secure:password" &&
+                     p.Value == "password").Should().BeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "url" &&
+                     p.Value == url.ToString()).Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "useAlternates" &&
+                     p.Value == "true").Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "teamcitySshKey" &&
+                     p.Value == "keyName").Should().NotBeNull();
+            properties.FirstOrDefault(
+                p => p.Name == "userNameStyle" &&
+                     p.Value == "AUTHOR_NAME").Should().NotBeNull();
```

Comment:
I know that this pattern FirstOrDefault().Should().NotBeNull() is used in other tests in FluentTc. I am afraid, that when the assertion fails, the error message would be something like: "something was null, but expected not to be null".
I suggest switching to 
```
properties.Should().ContainSingle(p => p.Name == "userNameStyle" && p.Value =="AUTHOR_NAME");

```

Reply:
Makes sense and makes for a more telling failure report.  Also updating the other test in this fixture.